src/app/features/auth/pages/registration/registration-page.component.ts

```diff
@@ -33,23 +39,18 @@ export class RegistrationPageComponent {
   ) {}
 
   public async register(credentials: Credentials): Promise<void> {
-    const [error] = await to<Register, HttpErrorResponse>(
-      firstValueFrom(
-        this._store
-          .dispatch(new Register(credentials))
-          .pipe(catchError((e: HttpErrorResponse) => Promise.reject(e)))
-      )
+    const [error] = await to<unknown, HttpErrorResponse>(
+      firstValueFrom(this._store.dispatch(new Register(credentials)))
     )
 
     if (error) {
       const registerError = error.error as ErrorResponse
 
-      this._toastr.error(registerError.message, registerError.error)
-
-      return
+      return this._toastr.error(registerError.message, registerError.error)
     }
     this._toastr.success('Registration successful!', 'Success')
-    await this._router.navigate(['/auth/login'], {
+
+    return void this._router.navigate(['/auth/login'], {
       relativeTo: this._route.parent
```

Comment:
Try to avoid specifying this. Maybe you cannot remove it entirely, but at least optimize it.